app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,66 @@
+require "gds_api/account_api"
+
+class SessionsController < ApplicationController
+  include AccountConcern
+
+  before_action :set_no_cache_headers
+
+  def create
+    redirect_with_ga account_manager_url and return if logged_in?
+
+    redirect_with_ga GdsApi.account_api.get_sign_in_url(
+      redirect_path: params[:redirect_path],
+      state_id: params[:state],
+    ).to_h["auth_uri"]
+  end
+
+  def callback
+    redirect_to Plek.new.website_root and return unless params[:code] && params[:state]
+
+    callback = GdsApi.account_api.validate_auth_response(
+      code: params.require(:code),
+      state: params.require(:state),
+    ).to_h
+
+    set_account_session_header(callback["govuk_account_session"])
+
+    redirect_with_ga(callback["redirect_path"] || account_manager_url, callback["ga_client_id"])
```

Comment:
I know this is how it currently is in finder-frontend, but this could be

```ruby
redirect_with_ga(callback.fetch("redirect_path", account_manager_url), callback["ga_client_id"])
```

Up to you.

Reply:
yeah great, i prefer the fetch default, thanks!

Reply:
So i tried this there's a catch. The callback is returning:
`{"govuk_account_session"=>"govuk-account-session", "redirect_path"=>nil, "ga_client_id"=>nil} `

This is because we're passing it no values here in the mock:
[stub_account_api_validates_auth_response(redirect_path: nil, ga_client_id: nil)](https://github.com/alphagov/frontend/pull/2668/files#diff-e9270928c40e1eef08f8c8576f6c7d9f774a7089b3c5ac515da590f2952bca02R71)

Which we decided to do here, because there is a [sensible default in the GDS API Adaptor test helper](https://github.com/alphagov/gds-api-adapters/blob/e0ed4c8070a2b37009be2cdcb735564f82d7b7c7/lib/gds_api/test_helpers/account_api.rb#L17).

I think our options are:
1. leave this unchanged and not use fetch. Accept but ignore the crumminess in the tests
2. call `.compact` [in the test helper ](https://github.com/alphagov/gds-api-adapters/blob/e0ed4c8070a2b37009be2cdcb735564f82d7b7c7/lib/gds_api/test_helpers/account_api.rb#L22). This seems more elegant but I'd want to think through if there are any scenarios where it would be valid to have a key with no string value... I believe there are some header scenraios where that's possible?
3. I could also put this in the controller code, but that seems to have all the issues of above plus we're changing our code to suit our tests. would prefer to avoid this one.


I can call `.compact` on it, but the

Reply:
For now, it's unchanged

Reply:
Ok, let's leave it unchanged